airflow/cli/commands/kubernetes_command.py

```diff
@@ -108,11 +113,15 @@ def cleanup_pods(args):
             pod_phase = pod.status.phase.lower()
             pod_reason = pod.status.reason.lower() if pod.status.reason else ''
             pod_restart_policy = pod.spec.restart_policy.lower()
+            current_time = datetime.now(pod.metadata.creation_timestamp.tzinfo)
+            # set time limit to 30m for pending pods
+            max_pending_time = timedelta(minutes=30)
```

Comment:
Should we turn the pending time into an optional parameter?

Reply:
@potiuk sure, addressed, PTAL. onething to remind is that I think we should cleanup old pending pods, otherwise we don't got a chance to do the deletion, as we have already filtered with labels, it's safe. And for newly pods, it might take time to be scheduled, so no cleanup action should be taken.

Reply:
Agree. But the 30 m is kinda arbitrary. And if you know it's safe, cleaning pods that are there for 10 minutes when you experienced some problematic case and want to clean-up might be useful I believe (especially if your 3 hours long meeting start in 5 minutes and you cannot wait ;) .